Remotion/Web/Development.Analyzers.IntegrationTests/RMWEB0003_TypeNotFromSameAssemblyWxeResourcePageStepAnalyzerTests.cs

```diff
@@ -0,0 +1,73 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using Verifier = Remotion.Web.Development.Analyzers.IntegrationTests.CSharpAnalyzerVerifier<Web.Development.Analyzers.RMWEB0003TypeNotFromSameAssemblyWxeResourcePageStepAnalyzer>;
+
+namespace Remotion.Web.Development.Analyzers.IntegrationTests
+{
+  [TestFixture]
+  public class RMWEB0003TypeNotFromSameAssemblyWxeResourcePageStepAnalyzerTests
+  {
+    [Test]
+    public async Task TypeFromOtherAssembly_Invalid ()
+    {
+      const string input = """
+                           using System.Text;
+                           using Remotion.Web.ExecutionEngine;
+                           public class A
+                           {
+                             public void Test()
+                             {
+                               new WxeResourcePageStep(typeof(int), "Test.aspx");
+                             }
+                             
+                             public WxeResourcePageStep Step => new(typeof(string), "Test.aspx");
+                           }
+                           """;
+
+      var diagnostic1 = Verifier.Diagnostic()
+          .WithSpan(7, 5, 7, 54)
+          .WithMessage("'int' is not from the same assembly as the containing type 'A'. This might be a mistake.");
```

Comment:
"Type 'int' is not defined in the same assembly as the containing type 'A'. If this is intentional, you can suppress the diagnostic ."